src/db/car/forest.rs

```diff
@@ -84,21 +89,21 @@ pub struct ForestCar<ReaderT> {
     // Multiple `ForestCar` structures may share the same cache. The cache key is used to identify
     // the origin of a cached z-frame.
     cache_key: CacheKey,
-    indexed: CarIndex<ReaderT>,
+    indexed: index::Reader<positioned_io::Slice<ReaderT>>,
     frame_cache: Arc<Mutex<ZstdFrameCache>>,
     write_cache: Arc<RwLock<ahash::HashMap<Cid, Vec<u8>>>>,
     roots: Vec<Cid>,
 }
 
 impl<ReaderT: super::RandomAccessFileReader> ForestCar<ReaderT> {
-    pub fn new(reader: ReaderT) -> io::Result<Self> {
+    pub fn new(reader: ReaderT) -> io::Result<ForestCar<ReaderT>> {
         let (header, footer) = Self::validate_car(&reader)?;
 
-        let index = CarIndex::open(reader, footer.index)?;
+        let indexed = index::Reader::new(positioned_io::Slice::new(reader, footer.index, None))?;
```

Comment:
The `reader` here can be of `EitherMmapOrRandomAccessFile` type, have you verifyied that `positioned_io::Slice` works fine with `mmap2` with no performance panity?

Reply:
It should, positioned_io::Slice simply does an addition on each read_at call